client/dashboard/domains/overview-glue-records/edit.tsx

```diff
@@ -0,0 +1,73 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { notFound, useRouter } from '@tanstack/react-router';
+import { useDispatch } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import {
+	domainGlueRecordsQuery,
+	domainGlueRecordUpdateMutation,
+} from '../../app/queries/domain-glue-records';
+import { domainRoute, domainGlueRecordsRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import DomainGlueRecordsForm from './form';
+import type { FormData } from './form';
+
+export default function EditDomainGlueRecords() {
+	const router = useRouter();
+	const { domainName, nameServer } = domainRoute.useParams();
+	const { data: glueRecordsData } = useQuery( domainGlueRecordsQuery( domainName ) );
```

Comment:
Since we're already using loading this query in the router we can use React Suspense here. That way we don't have to deal with the case where `glueRecordsData` is undefined.

```suggestion
	const { data: glueRecordsData } = useSuspenseQuery( domainGlueRecordsQuery( domainName ) );
```

Reply:
Yep, makes sense. Updated in https://github.com/Automattic/wp-calypso/pull/105213/commits/8002ca0846c82ad75e0b6838e045b245e629c439.